### ISSUE TITLE
Typing: CodeField component requires optional props of TextInput

### ISSUE THREAD

**Issue by alexstrat** [retyui/react-native-confirmation-code-field#126]
With the example from the Readme:
```tsx
        <CodeField
          ref={ref}
          {...props}
          value={codeValue}
          autoFocus
          selectionColor={COLORS.BUTTON_INACTIVE}
          onChangeText={onCodeValueChanged}
          cellCount={INPUT_CELL_COUNT}
          rootStyle={styles.codeFiledRoot}
          keyboardType="number-pad"
          renderCell={({ index, symbol, isFocused }) => (
            <Text
              key={index}
              style={[
                styles.codeVerificationInputCell,
                index >= 1 && styles.codeVerificationInputCellMargin,
              ]}
              onLayout={getCellOnLayoutHandler(index)}
            >
              {symbol || (isFocused && <Cursor />)}
            </Text>
          )}
        />
```

Got this typing error:

```
Type '{ value: string; autoFocus: true; selectionColor: COLORS; onChangeText: (value: string) => void; cellCount: number; rootStyle: { marginVertical: number; }; keyboardType: "number-pad"; renderCell: ({ index, symbol, isFocused }: { ...; }) => Element; ref: RefObject<...>; }' is missing the following properties from type 'Pick<TextInputProps, "hitSlop" | "onLayout" | "pointerEvents" | "removeClippedSubviews" | "testID" | "nativeID" | "collapsable" | "needsOffscreenAlphaCompositing" | ... 97 more ... | "textAlignVertical">': accessibilityValue, onAccessibilityEscape, passwordRules, rejectResponderTermination, importantForAutofillts(2739)
```

**Comment by retyui** [comment]
@alexstrat 

I've just created a react-native project based on typescript template nd check these issues

Everything works fine!

could you please create a demo repo ?

**Comment by alexstrat** [comment]
Thanks for your answer!

Actually, I think I found a track.

You should be able to reproduce the issue if you use an older version of react-native and of its typing (0.60.x in my case). The required props `accessibilityValue`, `onAccessibilityEscape`, `passwordRules`, `rejectResponderTermination`, `importantForAutofillts` were added on `TextInput` in [recent versions](https://github.com/react-native-community/releases/blob/master/CHANGELOG.md) of react-native.

Indeed, [in the npm package](https://unpkg.com/react-native-confirmation-code-field@6.0.3/lib/CodeField.d.ts), the typing of `CodeField`'s **picks** from `TextInputProps` every props of  `TextInputProps` (as of 0.62 version) except `style`:

```ts
declare const _default: React.ForwardRefExoticComponent<
  {
    rootStyle?: any;
    textInputStyle?: StyleProp<TextStyle>;
    ....
  } & Pick<
    TextInputProps,
    | 'value'
    | 'onBlur'
    | 'onFocus'
    ....etc
    | 'allowFontScaling'
    | 'autoCapitalize'
    | 'textAlignVertical'
  > &
    React.RefAttributes<TextInput>
>;
```

Though, because i'm using older version of react-native, some of the picked properties are not present on `TextInputProps`. And you [can verify that when you pick inexistent property on an interface, these properties become mandatory](https://www.typescriptlang.org/play/#code/C4TwDgpgBACgTgezAZygXigbwFBT1AMwQQH4AuKAOwFcBbAIwjgBpd8B6dqegQznKp1GcbAF9snKAAFgyALQBLAOaUEcCNlCRYCgMYBrCABN4SVBhh79AHlMpmUAOREEjqAB8nvOI4B82bF0ESmRgKDBEFApLA2M7cywxAG5sIA) — unfortunately.

Though what bugs me, is that in [source code **Omit** is used](https://github.com/retyui/react-native-confirmation-code-field/blob/master/src/CodeField.tsx#L26) and its seems like it gets transformed to **Pick** on compilation. If **Omit** was emitted, the problem would not appear.

**Comment by retyui** [comment]
@alexstrat 

Fixed in [6.1.0](https://github.com/retyui/react-native-confirmation-code-field/blob/master/CHANGELOG.md#610)
